packages/auth/src/Expo-AuthSession/ExpoLoginButton.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { Button } from "@equinor/mad-components";
+import * as WebBrowser from "expo-web-browser";
+import {
+    exchangeCodeAsync,
+    ResponseType,
+    useAuthRequest,
+    useAutoDiscovery,
+} from "expo-auth-session";
+import { MadAccount, UserInfo } from "../types";
+import { LoginButtonProps } from "../components";
+import { jwtDecode } from "jwt-decode";
+import "core-js/stable/atob";
+
+WebBrowser.maybeCompleteAuthSession();
+
+export const ExpoLoginButton = ({
+    clientId,
+    redirectUri,
+    onAuthenticationSuccessful,
+    onAuthenticationFailed,
+    enableAutomaticAuthentication,
+    scopes = [],
+    title = "Log in",
+    ...rest
+}: LoginButtonProps) => {
+    const discovery = useAutoDiscovery(
+        "https://login.microsoftonline.com/statoilsrm.onmicrosoft.com/v2.0",
+    );
+
+    const [request, , promptAsync] = useAuthRequest(
+        {
+            clientId,
+            scopes: scopes,
+            redirectUri,
+            responseType: ResponseType.Code,
+        },
+        discovery,
+    );
+
+    const onPress = () => {
+        promptAsync()
```

Comment:
I would strongly advise using the async await paradigm for working with async methods. Code like this is very hard to read (nested `then`'s and `catch`es)

Reply:
refactored function using await and surrounded in a single trycatch block